client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-set-favorite/index.tsx

```diff
@@ -0,0 +1,150 @@
+import { Gridicon, Button } from '@automattic/components';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { ReactElement, useContext, useState } from 'react';
+import { useQueryClient } from 'react-query';
+import { useDispatch } from 'react-redux';
+import useRemoveFavoriteSiteMutation from 'calypso/data/agency-dashboard/use-remove-favourite-site-mutation';
+import useSetFavoriteSiteMutation from 'calypso/data/agency-dashboard/use-set-favourite-site-mutation';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, successNotice, removeNotice } from 'calypso/state/notices/actions';
+import SitesOverviewContext from '../context';
+import type { APIError, Site } from '../types';
+
+import './style.scss';
+
+interface Props {
+	isFavorite: boolean;
+	siteId: number;
+	siteUrl: string;
+}
+
+export default function SiteSetFavorite( { isFavorite, siteId, siteUrl }: Props ): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const queryClient = useQueryClient();
+
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
```

Comment:
I've started noticing this pattern of using separate `isSubmitting` state for requests when using react-query.
React Query already provides this for us in the form of `.isLoading`:
```
const { isLoading } = useMutation();
const { isLoading, isFetching } = useQuery();
```

Since in this case, we have 2 mutations, either of which could be loading then we can use something like this:
```
const setFavorite = useSetFavoriteSiteMutation( handleMutation() );
const removeFavorite = useRemoveFavoriteSiteMutation( handleMutation() );
const isSubmitting = setFavorite.isLoading || removeFavorite.isLoading;
```

This way we avoid having extra state that:
- duplicates already existing state
- forces us to manually keep it in sync inside success and error handlers via setIsSubmitting

Site-note:
We can use a single mutation (e.g. `useToggleFavoriteSiteMutation`) that accepts an argument whether the favorite status should be set or removed which can simplify things.

Reply:
Happy to help :)